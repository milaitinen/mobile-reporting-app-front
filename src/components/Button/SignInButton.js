import React from 'react';
import { Text, TouchableHighlight } from 'react-native';

import styles from './styles';

const SignInButton = ({ onPress, children }) => (
    <TouchableHighlight
        onPress={onPress}
        style={styles.touchable}
    >
        <Text style={styles.text}>{children}</Text>
    </TouchableHighlight>
);

export default SignInButton;