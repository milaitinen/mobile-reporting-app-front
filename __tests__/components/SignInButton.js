import React from 'react';
import { shallow, configure } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import renderer from 'react-test-renderer';

import SignInButton from '../../src/components/Button/SignInButton';

configure({ adapter: new Adapter() });

it('Sign in button renders correctly', () => {
    const tree = renderer.create(
        <SignInButton onPress={jest.fn()}>
            Sign in
        </SignInButton>
    ).toJSON();
    expect(tree).toMatchSnapshot();
});

it('pressing the Sign in button works correctly', () => {
    const mockPress = jest.fn();
    const wrapper = shallow(<SignInButton onPress={mockPress}/>);
    wrapper.find('TouchableHighlight').simulate( 'press' );
    expect(mockPress).toHaveBeenCalled();
});