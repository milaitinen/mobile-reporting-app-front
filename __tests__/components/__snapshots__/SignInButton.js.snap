// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Sign in button renders correctly 1`] = `
<View
  style={undefined}
>
  <View
    accessibilityComponentType={undefined}
    accessibilityLabel={undefined}
    accessibilityTraits={undefined}
    accessible={true}
    hasTVPreferredFocus={undefined}
    hitSlop={undefined}
    isTVSelectable={true}
    nativeID={undefined}
    onLayout={undefined}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Array [
        Object {
          "backgroundColor": "transparent",
        },
        undefined,
      ]
    }
    testID={undefined}
    tvParallaxProperties={undefined}
  >
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      style={undefined}
    >
      Sign in
    </Text>
  </View>
</View>
`;
